### PR TITLE
Added basic hardware settings to frontend

### PR DESCRIPTION
**Pull request by 0phelia36**
## TL;DR
Users can now change the hardware settings for the microlab through the web ui, download existing configurations, and upload new ones. 

Configuration and data storage was also changed, configuration is now done in /etc/microlab rather than through config.py, and recipe and hardware configurations are stored in /var/lib/microlab

## What
What is affected by this PR?
- [X] API
- [X] GUI
- [X] Hardware Integration
- [X] OS/Deployment
- [ ] Documentation
- [ ] Other (please describe in notes)

## Testing
How was this tested?
- [ ] Locally Virtualized
- [X] Raspberry Pi
- [ ] With Hardware
- [ ] Other (please describe in notes)

## Notes
How was your day?

### CODE REVIEW

File: backend/config.py [hunk 1]
```diff
@@ -1,29 +1,141 @@
...
+                    section_list.append(key)
+                else:
+                    section_list.append('[missing section]')
+                section_string = '.'.join(section_list)
+                if error == False:
+                    error = 'Missing value or section.'
+                print("Configuration error at {0}: '{1}', falling back to default value '{2}'."
+                    .format(section_string, error, default))
 
+
```
Comment (sparklestheunicorn) [inline backend/config.py:50]: Love everything that is happening here.

File: gui/src/pages/Settings.jsx [hunk 2]
```diff
@@ -1,10 +1,26 @@
...
-	return (
-		<div>
-			<p>Settings page here...</p>
-		</div>
-	);
+  // dummy counter to have components refetch data
+  const [counter, updateCounter] = useReducer(x => x + 1, 0)
+
+  return (
+    <div className="settings-page">
```
Comment (sparklestheunicorn) [inline gui/src/pages/Settings.jsx:15]: :+1:

File: gui/src/pages/Settings.jsx [hunk 3]
```diff
@@ -1,10 +1,26 @@
...
 export function Settings() {
-	return (
-		<div>
-			<p>Settings page here...</p>
-		</div>
-	);
+  // dummy counter to have components refetch data
+  const [counter, updateCounter] = useReducer(x => x + 1, 0)
+
+  return (
```
Comment (sparklestheunicorn) [inline gui/src/pages/Settings.jsx:14]: Let's add the `page` class here to eliminate some redundant styles

File: gui/src/pages/Settings.scss [hunk 4]
```diff
@@ -0,0 +1,14 @@
+.settings-page {
+  overflow-y: scroll;
+  padding: 1rem;
+  display: block;
+  height: 260px;
+}
```
Comment (sparklestheunicorn) [inline gui/src/pages/Settings.scss:6]: This should probably get merged with the .page styles.

**Comment by sparklestheunicorn** [comment]
Only nitpicks are on some redundant style rules. Otherwise this looks good to me and I'm very excited about exposing this in the UI.
